cpu/cc2538/periph/uart.c

```diff
@@ -308,11 +185,64 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
 
 void uart_poweron(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
 
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 }
 
 void uart_poweroff(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
+
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 0;
 }
+
+void uart_clockon(uart_t uart)
+{
+    assert(uart < UART_NUMOF);
+
+    SYS_CTRL->cc2538_sys_ctrl_unnamed1.RCGCUART |= (1 << uart);
+}
+
+void uart_clockoff(uart_t uart)
+{
+    assert(uart < UART_NUMOF);
+
+    SYS_CTRL->cc2538_sys_ctrl_unnamed1.RCGCUART &= ~(1 << uart);
+}
+
+static inline void irq_handler(uart_t uart)
+{
+    assert(uart < UART_NUMOF);
+
+    uint_fast16_t mis;
+
+    /* Latch the Masked Interrupt Status and clear any active flags */
+    mis = uart_config[uart].dev->cc2538_uart_mis.MIS;
+    uart_config[uart].dev->ICR = mis;
+
+    while (uart_config[uart].dev->cc2538_uart_fr.FRbits.RXFE == 0) {
+        uart_ctx[uart].rx_cb(uart_ctx[uart].arg, uart_config[uart].dev->DR);
+    }
+
+    if (mis & (OEMIS | BEMIS | FEMIS)) {
+        /* ISR triggered due to some error condition */
+        reset(uart_config[uart].dev);
```

Comment:
I stumbled upon this in my last review, though I stayed quite. It seems to me, that the call to this reset function is broken in some regards:
- reset always enables RX mode, independent if it was enabled before (see my comment above)
- so we really need to clear the FIFOs on error? TX FIFO overrun is not possible in any case, and RX overrun is very unlikely, and even if it happens, I don't think we should throw away the already received data...

My approach here would be to remove the `reset()` function, and simply reset error bits

Reply:
Hmm yeah. So you are proposing just to discard the data byte and not re-initialise the hardware. I found [this discussion](https://electronics.stackexchange.com/questions/29134/handling-of-uart-errors) which basically says that you could pass these errors up to the higher level if we really want to ensure data integrity, but these errors are rare. Is there any case in which these errors can be generated by faulty hardware on this side? I don't know of any ways so my fixup just resets the error bits.